pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -493,77 +494,85 @@ func (e *Controller) syncService(key string) error {
 	}
 	subsets = endpoints.RepackSubsets(subsets)
 
-	// See if there's actually an update here.
-	currentEndpoints, err := e.endpointsLister.Endpoints(service.Namespace).Get(service.Name)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			currentEndpoints = &v1.Endpoints{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:   service.Name,
-					Labels: service.Labels,
-				},
+	var createEndpoints bool
+	var newEndpoints *v1.Endpoints
+
+	// Retry on conflict
+	err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		// See if there's actually an update here.
+		currentEndpoints, err := e.endpointsLister.Endpoints(service.Namespace).Get(service.Name)
```

Comment:
this gets the endpoints from the informer cache,

Reply:
True but if we do a get directly, it'll be expensive operation against apiserver. Either we need to requeue or do a get is what I am thinking.

Reply:
sorry, I didn't understand it at first , I thought you were trying to retry the Get :smile: 
I can see that the goal is to retry the Endpoint creation/update operation instead of requeueing.
I think that what you are trying to do is what @robscott solved in Endpoint Slices with https://github.com/kubernetes/kubernetes/pull/99345 .
However, I'd like to be conservative here and first have more evidence that this is a real problem, since the Endpoint Slices "achcey out of sync" issue was "frequently" spotted in the CI and we also were able to reproduced it with a test.